src/input/mobile_io.cpp

```diff
@@ -0,0 +1,99 @@
+#include "mobile_io.hpp"
+
+#include "ros/ros.h" // For ROS_WARN
+
+#include "hebi_cpp_api/lookup.hpp"
+#include "hebi_cpp_api/group_command.hpp"
+
+namespace hebi {
+
+MobileIODiff::MobileIODiff(const MobileIOState& prev, const MobileIOState& curr) {
+  for (size_t i = 0; i < NumButtons; ++i) {
+    if (prev.buttons_[i] && curr.buttons_[i])
+      buttons_[i] = ButtonState::On;
+    else if (!prev.buttons_[i] && curr.buttons_[i])
+      buttons_[i] = ButtonState::ToOn;
+    if (!prev.buttons_[i] && !curr.buttons_[i])
+      buttons_[i] = ButtonState::Off;
+    else if (prev.buttons_[i] && !curr.buttons_[i])
+      buttons_[i] = ButtonState::ToOff;
+  }
+}
+
+MobileIODiff::ButtonState MobileIODiff::getButton(int button) {
+  if (button < 1 || button > NumButtons) {
+    assert(false);
+    return ButtonState::Off;
+  }
+  return buttons_[button - 1];
+}
+
+std::unique_ptr<MobileIO> MobileIO::create(const std::string& family, const std::string& name) {
+  hebi::Lookup lookup;
+  std::shared_ptr<hebi::Group> group;
+  while (!group) {
+    ROS_WARN("Looking for Mobile IO Controller");
+    group = lookup.getGroupFromNames({ family }, { name });
+  }
+  return std::unique_ptr<MobileIO>(new MobileIO(group));
+}
+
+MobileIOState MobileIO::getState() {
+  bool tmp;
+  return getState(tmp);
+}
+MobileIOState MobileIO::getState(bool& got_feedback) {
```

Comment:
Only issue I have is that looks like there was a lot of copying and pasting without remembering to put a space between each function definition